### DISCUSSION TITLE
Feature Request: JSON file globbed parsing should provide the option to generate a column with the file name as CSV parsing does

### DISCUSSION THREAD

**Discussion by chrisfw** [duckdb/duckdb#6619] [Ideas]
CSV parsing provides the option for the parsed file name to be included as a column in the parse results.  This is important for multi-file processing and should be included in JSON parsing as well so that the source file can be identified in the generated table data.  If the json file data does not contain this info you have no way of identifying the source file name after the fact and would have to resort to processing the files individually which reduces the value and utility of the multi file processing feature.

e.g. create table sample as select * from read_json_auto('C:/temp/*.json.sql.gz',filename=true);

![duckdb_csv_parse_options](https://user-images.githubusercontent.com/6135741/223563606-193bfffc-b1a0-43c8-9137-936f45a88d9b.png)

**Comment by Tishj** [comment]
Duplicate of #6567 
But your interest is welcome 👍

**Comment by chrisfw** [comment]
Thank you for letting me know.

Regards,
Chris

**Comment by lnkuiper** [comment]
Closed via #7478
